JoyboyCommunity/src/context/Theme.tsx

```diff
@@ -1,16 +1,31 @@
-import {createContext, useMemo} from 'react';
-import {useColorScheme} from 'react-native';
+import {createContext, useMemo, useState} from 'react';
+import {ColorSchemeName, useColorScheme} from 'react-native';
 
 import {DarkTheme, LightTheme, Theme} from '../styles';
 
-export const ThemeContext = createContext<Theme>(LightTheme);
+export type IThemeContext = {
+  theme: Theme;
+  toggleTheme: () => void;
+  scheme: 'dark' | 'light';
+};
+
+export const ThemeContext = createContext<IThemeContext>({
+  theme: LightTheme,
+  toggleTheme: () => {},
+  scheme: 'light',
+});
 
 export const ThemeProvider: React.FC<React.PropsWithChildren> = ({children}) => {
   const defaultColorScheme = useColorScheme();
 
   const colorScheme = defaultColorScheme;
+  const [scheme, setScheme] = useState<'dark' | 'light'>('light');
```

Comment:
Default value for the state should be the `defaultColorScheme `

Reply:
It's like an enum string of the theme mode we have. Why we need to change it?
Also don't know how we can change the defaultColorScheme as default for user (maybe we can do a localstorage after)

Reply:
I meant it should be
```ts
const [scheme, setScheme] = useState<'dark' | 'light'>(defaultColorScheme);
```